### PR TITLE
Lazy loading implementation

### PR DESCRIPTION
**Pull request by Kajol-Kumari**
This PR includes `complete lazy loading implementation` and all modules refactoring.

Working GIF of feature:
ezgif com-video-to-gif

### CODE REVIEW

**Comment by Kajol-Kumari** [comment]
@Sanji515 @RishabhJain2018 @lunayach @Shekharrajak I request you to please review this PR as i then need to modify the other PRs accordingly.

File: src/app/shared/common-shared.module.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { NgModule } from '@angular/core';
```
Comment (Sanji515) [inline src/app/shared/common-shared.module.ts]: Good job :+1: @Kajol-Kumari Thats what I was thinking when I was reviewing the previous PR of lazy loading to have common module as we were having too many common imports

Comment (Sanji515) [inline src/app/shared/common-shared.module.ts]: But can we change the name of file to just `shared.module.ts` or change the folder name to `common` and then file name to `common.module.ts`.

**Review by Sanji515 — COMMENTED** [review]
Everything is working fine :+1:

File: src/app/components/publiclists/publiclist.module.ts [hunk 2]
```diff
@@ -1,56 +1,44 @@
...
-import { ChallengeService } from '../../services/challenge.service';
-
 // import routes
-import { PubliclistRoutingModule } from './publiclist-routing.module';
+import { PubliclistRoutingModule, TeamlistsRoutingModule } from './publiclist-routing.module';
 
 // import Module
-import { NavModule } from '../nav/nav.module';
+import { ChallengelistModule } from './challengelist/challengelist.module';
+import { CommonSharedModule } from '../../shared/shared.module';
```
Comment (Sanji515) [inline src/app/components/publiclists/publiclist.module.ts]: Also can we import it as `SharedModule` @Kajol-Kumari

**Review by Sanji515 — APPROVED** [review]
LGTM! This can be merged :tada:

**Pull request merged**